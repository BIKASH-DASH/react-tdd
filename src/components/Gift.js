import React, { Component } from 'react'
import {FormControl,Form,FormGroup,Button,FormLabel} from 'react-bootstrap'
export default class Gift extends Component {
    constructor(){
        super();
        this.state = {
            person:'',
            present:''
        }
    }
    render() {
        return (
            <div>
                <Form>
                    <FormGroup>
                        <FormLabel>Person Name</FormLabel>
                        <FormControl 
                        className='input-person'
                        onChange={(e)=>this.setState({person:e.target.value})}
                        />
                    </FormGroup>
                    <FormGroup>
                        <FormLabel>
                            present
                        </FormLabel>
                        <FormControl onChange={(e)=>this.setState({present:e.target.value})} className='input-present'/>
                    </FormGroup>
                    <Button onClick={()=>this.props.removeGift(this.props.gift.id)} className='remove-Gift'>Remove Gift</Button>
                </Form>
            </div>
        )
    }
}
