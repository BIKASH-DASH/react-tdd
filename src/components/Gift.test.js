import React from 'react';
import Enzyme,{ shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import Gift from './Gift';
Enzyme.configure({ adapter: new Adapter() });
const wrapper = shallow(<Gift/>);


describe('Gift component', () => {

    it('should render properly', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
})
