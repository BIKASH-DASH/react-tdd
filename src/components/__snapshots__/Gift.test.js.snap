// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift component should render properly 1`] = `ShallowWrapper {}`;
